framework/src/kernels/ArrayKernel.C

```diff
@@ -102,9 +103,10 @@ ArrayKernel::computeResidual()
     initQpResidual();
     for (_i = 0; _i < _test.size(); _i++)
     {
-      _work_vector = computeQpResidual() * _JxW[_qp] * _coord[_qp];
+      computeQpResidual(_work_vector);
```

Comment:
You probably want to zero `_work_vector` before, no?

Reply:
Do you mean in this loop or in the constructor? Seems like unnecessary operation in this loop.

Reply:
Someone could easily add to the vector instead of setting it.

Reply:
You're right. I'll add a `setZero` then. Seems silly that someone would just add to it before knowing what is in the vector, but it is very possible.

Reply:
Yeah - I definitely agree. Very unlikely but could easily make one's day horrid 💩